project_sla/project_sla_control.py

```diff
@@ -320,9 +325,21 @@ class SLAControlled(orm.AbstractModel):
     }
 
     def store_sla_control(self, cr, uid, ids, context=None):
-        docs = self.browse(cr, uid, ids, context=context)
-        self.pool.get('project.sla.control').store_sla_control(
-            cr, uid, docs, context=context)
+        if ids:
+            try:
+                cr.execute("SAVEPOINT %s_sla_save" % self._table)
+                cr.execute("""
+                    SELECT * FROM %s WHERE id IN %%s
+                    FOR UPDATE of %s NOWAIT
+                """ % (self._table, self._table), (tuple(ids),))
+            except:
+                cr.execute("ROLLBACK TO %s_sla_save" % self._table)
+                _logger.error("Cannot lock table %s", self._table)
+                raise
```

Comment:
It was more that a year ago, when I was playing with this module, but as I remember, it had a very slow performance on mass updates, when write some field on all documents of model, trigger recalculation of all bound SLA lines, which takes too much time.

And there was situation when half of SLA lines was recomputed, but transaction breaks because one of documents that have not been processed yet, was updated or removed.

It was too long time ago, so I do not remember correct reasons for this code, but now it seems working, and at least this code is running and do not takes my attenting now.

To simplify this PR, i can remove this piece of code...